yamllint/linter.py

```diff
@@ -227,7 +227,7 @@ def run(input, conf, filepath=None):
 
     if isinstance(input, (bytes, str)):
         return _run(input, conf, filepath)
-    elif hasattr(input, 'read'):  # Python 2's file or Python 3's io.IOBase
+    elif hasattr(input, 'read'):  # isinstance(input, io.IOBase)
```

Comment:
What do you think about:
```suggestion
    elif isinstance(input, io.IOBase)
```
now that yamllint only supports Python 3?

Reply:
Sure. I just thought that you might want to support optional types with a `read` method, but no `io.IOBase` inheritance. That could make sense in a general purpose library, but probably not here.

Reply:
`io.IOBase` is supposed to be [the abstract base class for all I/O classes](https://docs.python.org/3/library/io.html#io.IOBase), so these hypothetical *optional types* should theoretically derive from it (I hope they do!)